doc_internal/support-tiers.md

```diff
@@ -0,0 +1,119 @@
+# "Support" Tierlist
+
+This document is guidance on what platforms and terminals fish supports to what degree.
+
+It is supposed to explain to users to what degree they can expect fish to work, and to help us decide when features impact compatibility.
+
+In an ideal world, everything would work everywhere. But of course, we have limited resources and need to not only decide where to put them,
+but also see where they currently *happen to be*. And there are times where we need to decide between supporting most users better,
+and supporting a small group of users. This may even mean dropping platforms or terminals when we can't make them work without impacting
+more important ones.
+
+It is also a list of targets we want to check before release.
+
+For the purposes of this list, terminal multiplexers like tmux and other forms of terminal-in-a-terminal like mc are "terminals".
+
+None of this is a legal document, and fish remains developed by hobbyists in their free time. Nobody is allowed to call me up at night to fix any of this.
+
+## The Tiers
+
+1. Working git at all times - can you expect a random checkout to be usable on this particular platform/terminal?
+   If this is broken, we should revert the offending commits ASAP.
+2. Tested before release, we don't want to ship with it broken
+3. We hope this works and will try to fix it if it doesn't, but no guarantees
+4. We don't care. If it works, that's cool.
+
+These tiers depend on how important a given target is to us, and how much information we get about it.
+
+That means, a tier 1 target requires a core developer to run it in anger, and working CI for operating systems/architectures.
```

Comment:
in anger?

Reply:
[possibly this idiom does not translate well](https://english.stackexchange.com/a/30943/99151)

Reply:
We could rephrase this as "regularly". What I mean to say is that people who are around use that target so regularly that we're likely to catch regressions within roughly "a day" (or possibly week, it's all a bit wishy washy).